question_6.py

```diff
@@ -0,0 +1,74 @@
+from typing import Any
+
+import requests
+from question_5 import WeatherForecaster
+
+
+class Forecaster(WeatherForecaster):
+    def __init__(self) -> None:
+        self.__CURRENT_WEATHER_ROUTE = "/forecast.json"
+        WeatherForecaster.__init__(self)
+
+    def __fetch(self, url: str) -> dict:
```

Comment:
any specific reason to use double underscore before the function name? and in `Forecaster` class you override __fetch function

Reply:
used double under score to make this function private, so that user only has access to get_forecast_data().
Yes I did override the __fetch since. WeatherForecaster.__fetch filter data differently and I needed a different implementation here.